### ISSUE TITLE
suggested hyper parameter - LightGBM

### ISSUE THREAD

**Issue by Chuang1128** [microsoft/LightGBM#6050]
I have 4,200,000 pieces of data with 20 features each. 
The models trained on this data all have a training accuracy of 0.99. However, when testing on other data, the accuracy drops to around 0.3 or 0.5. There's an overfitting issue. 
How should I configure my hyperparameters? 

I've looked into the suggested recommendations below, but I'm still unable to achieve improvements.
Deal with Over-fitting
- Use small max_bin

- Use small num_leaves

- Use min_data_in_leaf and min_sum_hessian_in_leaf

- Use bagging by set bagging_fraction and bagging_freq

- Use feature sub-sampling by set feature_fraction

- Use bigger training data

- Try lambda_l1, lambda_l2 and min_gain_to_split for regularization

- Try max_depth to avoid growing deep tree

- Try extra_trees

- Try increasing path_smooth

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

With just the information you've provided, it's not possible to answer "How should I configure my hyperparameters?". The best hyperparameters for training a supervised learning model are highly dependent on details you haven't shared here, like the task (i.e. loss function), distribution of the target and distributions of the features.

Try comparing the effect of different sets of hyperparameters using cross-validation. This guide from `scikit-learn` is very good, even if you aren't using `scikit-learn` specifically: https://scikit-learn.org/stable/modules/cross_validation.html.

You might also find this useful: https://github.com/microsoft/LightGBM/issues/6044#issuecomment-1683052078

> I've looked into the suggested recommendations below

Looks like you got this list from https://lightgbm.readthedocs.io/en/latest/Parameters-Tuning.html#deal-with-over-fitting. That's a great place to start! In the future, when you report things copied from documentation here, please do link to that documentation. That'll help people to understand where the things you're saying came from.

**Comment by Chuang1128** [comment]
Understood, I will pay attention in the future.
I would like to know what are the typical ranges for the hyperparameters of LightGBM

**Comment by jameslamb** [comment]
> I would like to know what are the typical ranges for the hyperparameters of LightGBM

As I said, that's "highly dependent on details you haven't shared here, like the task (i.e. loss function), distribution of the target and distributions of the features".

**Comment by Chuang1128** [comment]
My loss function is defined by the following formula, with the target variables and feature distribution being continuous.
1-sum((Obs-Pred)^2)/sum((Obs-mean(Obs))^2)

**Comment by jameslamb** [comment]
That is just not enough information for us to suggest ranges for you. Please use cross validation to explore ranges of hyperparameters for your application. The links I provided above should be sufficient to get you started if you're using Python (you have yet to say how you're using LightGBM). Similar approaches are available in other languages as well.
